### PR TITLE
changing the door of escape-game to a portal

### PR DESCRIPTION
**Pull request by anj226**

### CODE REVIEW

File: extensions/escape-game/standalone.mjs [hunk 1]
```diff
@@ -161,31 +171,59 @@ class Standalone {
...
-        name: LAYER_DOOR.layerName,
-        cells: []
-      }),
-    );
-    setTimeout(async function(obj) {
-      if (--obj.doorOpenPressedCount == 0) {        
-        await obj.helper.broadcastCellSetUpdateToAllUser(
+    const mapName = "world1";
+    if (++this.doorOpenPressedCount === 1) {  
+      console.log("open!");
```
Comment (john0312) [inline extensions/escape-game/standalone.mjs:176]: Maybe `escape-room door is open due to e2s call.`?

File: extensions/escape-game/standalone.mjs [hunk 2]
```diff
@@ -161,31 +171,59 @@ class Standalone {
...
+      return;
+    }
+    if (mapCoord === 'undefined') {
+      // No location? Just skip.
+      return;
+    }
+    const explodeCell = this.helper.gameMap.getCell(mapCoord, LAYER_DOOR.layerName);
+    if (explodeCell) { // if walk into opening door => teleport to somewhere
+      const target = mapCoord.copy();
+      target.x = this.doorTarget['x'];
```
Comment (john0312) [inline extensions/escape-game/standalone.mjs:220]: You'll need to set the world name/map name, or the player will end up in the wrong place.

File: extensions/escape-game/standalone.mjs [hunk 3]
```diff
@@ -120,31 +120,41 @@ class Standalone {
       );
     });
     if (config.has('escape-game.doorPosition')) {
-      this.doorCells = [config.get('escape-game.doorPosition')];
+      this.doorCells = config.get('escape-game.doorPosition');
```
Comment (john0312) [inline extensions/escape-game/standalone.mjs:123]: Consider auto populating `this.doorCells[i]['idx']`.

I would recommend updating the config in hitcon-online too. Update the map if you've time, or leave a TODO.

File: extensions/escape-game/standalone.mjs (hunk 1 above)
Comment (anj226) [reply extensions/escape-game/standalone.mjs]: It is just my debug message ...... I think I will just remove it.

File: extensions/escape-game/standalone.mjs [hunk 4]
```diff
@@ -161,31 +171,59 @@ class Standalone {
...
-      "world1",
-      CellSet.fromObject({
-        name: LAYER_DOOR.layerName,
-        cells: []
-      }),
-    );
-    setTimeout(async function(obj) {
-      if (--obj.doorOpenPressedCount == 0) {        
-        await obj.helper.broadcastCellSetUpdateToAllUser(
+    const mapName = "world1";
```
[the remark this answers is not in the document]
Comment (anj226) [reply extensions/escape-game/standalone.mjs]: Fixed.

File: extensions/escape-game/standalone.mjs [hunk 5]
```diff
@@ -161,31 +171,59 @@ class Standalone {
...
+            name: `DOOR_${doorCell["idx"]}`,
+            cells: [doorCell]
           }),
-        );  
+        );
+      }
+    }
+    setTimeout(async function(obj) {
+      if (--obj.doorOpenPressedCount === 0) {
+        console.log("close!");
```
[the remark this answers is not in the document]
Comment (anj226) [reply extensions/escape-game/standalone.mjs]: It is just my debug message ...... I think I will just remove it.

**Review by john0312 — APPROVED** [review]

**Comment by john0312** [comment]
LGTM, but you might need to rebase.

**Review by john0312 — APPROVED** [review]

**Pull request merged**